### ISSUE TITLE
Error with listOpenTypeFeatures()

### ISSUE THREAD

**Issue by peternowell** [schriftgestalt/DrawBotGlyphsPlugin#4]
I recently upgraded to Big Sur (still using Glyphs 2.6.5). I rely on a script that uses DrawBot, and the `listOpenTypeFeatures()` function now consistently fails with a `KeyError: u'CTFeatureTypeIdentifier'`. It fails for nearly every font I try. I've been debugging this issue for days, and finally figured out that it only occurs when I run the script in Glyphs (Macro Panel or plugin DrawBot window). If I run the script in DrawBot, there's no problem.

If you'd like, here's a basic version of the code to test with (replace font file path):
```python
from drawBot import *
fontFile = 'path/to/font.otf' # replace with your own OTF or TTF font file
print( listOpenTypeFeatures( fontFile ) )
```

One of the fonts that seems to work without an error is [Ancho](http://beatrizl.com/project/ancho), although `listOpenTypeFeatures()` provides different results in Glyphs and in the latest version of DrawBot.
_Glyphs output for Ancho:_
`{'hwid': True}`
_DrawBot output for Ancho:_
`['kern', 'mark', 'mkmk']`

Could it be that the DrawBot framework has been updated but the Glyphs plugin hasn't incorporated it yet? Please help—I can't use this critical script without OpenType feature detection.

**Comment by jenskutilek** [comment]
This part of the code has been updated in newer DrawBot versions. I patched that specific part from the current DrawBot, though I don't have Big Sur here to test if everything else works.

The main problem is that Glyphs 2 is using Python 2, while the current DrawBot uses Python 3, so it is expected that more things might break as code is updated. It would be a lot of work to backport DrawBot updates to Python 2.

Glyphs 3 uses Python 3, so it will be easier to keep the DrawBot plugin updated for Glyphs 3.

**Comment by peternowell** [comment]
Thanks, Jens! That fixed the `listOpenTypeFeatures()` function, but setting features with `openTypeFeatures()` produces an error: `AttributeError: 'list' object has no attribute 'update'`. I'm guessing some part of that function expected the features list to be stored as a dictionary, which it used to be, instead of the current list. Should I open a new issue about this? 

Simply trying this code can reproduce the error:
```python
from drawBot import *
openTypeFeatures( liga=False )
```

**Comment by jenskutilek** [comment]
I've updated some more code, could you try again?

**Comment by peternowell** [comment]
Fixed! The `openTypeFeatures()` function now works as expected.

I also discovered another easily-correctable issue: the `append()` method on the `FormattedString` object no longer supports basic <str> strings—the content must be a unicode or FormattedString type. I know that Python 2 vs. 3 issues like this are probably numerous, but I hope we can address them as they appear. It would be a shame if DrawBot became unusable with Glyphs 2 so soon after 3 is released. After a year or two, I could understand.

Anyway, thank you so much for your help, Jens!
